### PR TITLE
Allow allow_blob_public_access and min_tls_version on azurerm_storage_account in USGovernmentCloud

### PR DESCRIPTION
**Pull request by ruandersMSFT**

### CODE REVIEW

**Comment by ruandersMSFT** [comment]
Ready for pull.  Has been tested to validate now works with Public and USGov Cloud.  Screenshot provided in issue to validate these features are available is US Gov region.

**Comment by ruandersMSFT** [comment]
This remains ready for pull, but you've added new pipeline steps that don't work.  Please pull ASAP so we can proceed with TLS1_2 deployments in MAG.

File: CHANGELOG.md [hunk 1]
```diff
@@ -7,6 +7,7 @@ FEATURES:
 BUG FIXES:
 
 * `azurerm_signar_service` - fixing a bug when expanding the `cors` block [GH-9671]
+* `azurerm_storage_account` - fixing an issue where `allow_blob_public_access` and `min_tls_version` are now available is US Government Cloud [GH-9128]
```
Comment (katbyte) [inline CHANGELOG.md:10]: We update the changelog by hand after merge so could we remove this?

File: azurerm/internal/services/storage/resource_arm_storage_account.go [hunk 2]
```diff
@@ -1131,10 +1137,12 @@ func resourceArmStorageAccountRead(d *schema.ResourceData, meta interface{}) err
...
 		d.Set("is_hns_enabled", props.IsHnsEnabled)
 		d.Set("allow_blob_public_access", props.AllowBlobPublicAccess)
-		// For all Clouds except Public, "min_tls_version" is not returned from Azure so always persist the default values for "min_tls_version".
+		// For all Clouds except Public and USGovernmentCloud, "min_tls_version" is not returned from Azure so always persist the default values for "min_tls_version".
 		// https://github.com/terraform-providers/terraform-provider-azurerm/issues/7812
 		// https://github.com/terraform-providers/terraform-provider-azurerm/issues/8083
-		if meta.(*clients.Client).Account.Environment.Name != autorestAzure.PublicCloud.Name {
+		// USGovernmentCloud "min_tls_version" allowed as of issue 9128
+		// https://github.com/terraform-providers/terraform-provider-azurerm/issues/9128
+		if meta.(*clients.Client).Account.Environment.Name != autorestAzure.PublicCloud.Name && meta.(*clients.Client).Account.Environment.Name != autorestAzure.USGovernmentCloud.Name {
```
Comment (katbyte) [inline azurerm/internal/services/storage/resource_arm_storage_account.go:1145]: could we pull out env name like above:
```suggestion
		// https://github.com/terraform-providers/terraform-provider-azurerm/issues/9128
		envName := meta.(*clients.Client).Account.Environment.Name
		if  envName != autorestAzure.PublicCloud.Name && envName != autorestAzure.USGovernmentCloud.Name {
```

**Review by katbyte — CHANGES_REQUESTED** [review]
Hi @ruandersMSFT - thanks for the PR. I've left a couple minor comments inline that need to be addressed. In addition to those could we remove the .vs files? thanks!

File: CHANGELOG.md (hunk 1 above)
Comment (ruandersMSFT) [reply CHANGELOG.md]: Removed, thanks.

File: azurerm/internal/services/storage/resource_arm_storage_account.go (hunk 2 above)
Comment (ruandersMSFT) [reply azurerm/internal/services/storage/resource_arm_storage_account.go]: Env Name pulled out to variable.

**Comment by ruandersMSFT** [comment]
@katbyte Thanks, requested changed fixed, ready for approval.

**Comment by ruandersMSFT** [comment]
@katbyte Response was provided.  Ready for pull.

**Review by katbyte — APPROVED** [review]
Thanks @ruandersMSFT - LGTM 👍

**Pull request merged**